### DISCUSSION TITLE
Does the Avalonia Extension for VS2022 support synchronization between the code editor and the visual designer?

### DISCUSSION THREAD

**Discussion by anose001** [AvaloniaUI/Avalonia#16080] [Q&A]
When one clicks on an element in the XAML code, the corresponding element in the visual designer is highlighted.
Similarly, selecting an element in the visual designer will highlight the corresponding XAML code.
I have not been able to make this behavior happening in AXAML code.

Have anyone been using the workflow? Any suggestion for alternatives?

**Comment by thevortexcloud** [comment]
Avalonia does not have a designer. It's only a previewer. The previewer works by actually running your app with a special platform and just rendering the contents to the previewer surface (you can actually see this if you make your design time view model try to load real data).

The closest you are going to get this functionality (which I would argue is even better) is dev tools. Which you can open by pressing F12 if you have the package installed while your app is running. It works kind of like a web browser's dev tools work. Which means you can change values for things while the app is running to see how it would actually look.

**Reply by anose001** [reply]
I see, that is neat indeed. I guess for navigational purposes there must be other sort of workflows for finding elements in your axaml code and if not then I guess you godda be thinking strongly about how to properly structure the axaml file and your UI

**Reply by thevortexcloud** [reply]
It certainly takes a bit of practice, but after awhile you will probably be able to sort of visualise the document layout in your head, and also know how all the pieces should fit together to achieve the layout you want. Which I find easier than any XAML designer I have ever used. 

That said the Avalonia dev tools does show the hierarchy of the view. It just does not show the XAML for it since at that point it has all been compiled/parsed.
